simply/market.py

```diff
@@ -16,10 +23,16 @@ class Market:
 
     This class provides a basic matching strategy which may be overridden.
     """
-
-    def __init__(self, time, network=None, grid_fee_matrix=None):
+    def __init__(self, scenario=None, network=None, grid_fee_matrix=None):
         self.orders = pd.DataFrame(columns=Order._fields)
-        self.t = time
+        self.scenario = scenario
+
+        if self.scenario is not None:
+            if self.scenario.market is not None:
+                warnings.warn("Existing market in Scenario is overwritten with new market")
+            self.scenario.market = self
+            self.scenario.environment.get_grid_fee = self.get_grid_fee
+
```

Comment:
It does not seem necessary that scenario is part of the market. Let the scenario's make the link between market and environment. Also it is much clearer to understand the market as part of the scenario only and not vice versa.

Reply:
for time step information: this can be handled by a `set_timestep` function which is triggered by environment update or by adding the environment